### PR TITLE
Upgrade sdk to v0.196.0

### PR DESCRIPTION
**Pull request by marefr**
https://github.com/grafana/grafana-plugin-sdk-go/releases/tag/v0.196.0

### CODE REVIEW

**Comment by papagian** [comment]
> @papagian sry please disregard I had to sync with main

it seems that the enterprise specification got un-synced sometime ago (currently there is no CI step for enterprise similar to what we have in OSS); this inconsistency got fixed by #79017

**Review by wbrowne — APPROVED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -28,6 +28,9 @@ replace github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.
 // This will not be required in the next k8s release @v28-with-hook
 replace k8s.io/kube-openapi v0.0.0-20230717233707-2695361300d9 => github.com/ryantxu/kube-openapi v0.0.0-20231113051506-80b8e1dfdde6 // @grafana/grafana-app-platform-squad
 
+// Workaround for https://github.com/grafana/thema/blob/067284b336bbbc82d6549748e20143e2475a4989/internal/deepmap/oapi-codegen/pkg/codegen/codegen.go#L206-
```
[the remark this answers is not in the document]
Comment (marefr) [reply go.mod:31]: Yes. That code fails with github.com/getkin/kin-openapi@v0.122.0 and I don't really want to update that (now).

[the remark this answers is not in the document]
Comment (marefr) [reply go.mod:31]: Apparently this doesn't work since now there's a failure in the SDK instead so need to upgrade thema :scream:

File: go.mod [hunk 2]
```diff
@@ -507,3 +507,5 @@ replace xorm.io/xorm => ./pkg/util/xorm
 replace github.com/prometheus/alertmanager => github.com/grafana/prometheus-alertmanager v0.25.1-0.20231027171310-70c52bf65758
 
 exclude github.com/mattn/go-sqlite3 v2.0.3+incompatible
+
+replace github.com/grafana/thema => github.com/grafana/thema v0.0.0-20231204151833-93b6494d445e
```
Comment (andresmgot) [inline go.mod:511]: is this replace still needed?

Comment (marefr) [reply go.mod]: No good spotted. Removed

File: go.mod [hunk 3]
```diff
@@ -28,6 +28,9 @@ replace github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.
 // This will not be required in the next k8s release @v28-with-hook
 replace k8s.io/kube-openapi v0.0.0-20230717233707-2695361300d9 => github.com/ryantxu/kube-openapi v0.0.0-20231113051506-80b8e1dfdde6 // @grafana/grafana-app-platform-squad
 
+// The v0.120.0 is needed for now to be compatible with grafana/thema.
+replace github.com/getkin/kin-openapi => github.com/getkin/kin-openapi v0.120.0
```
Comment (sunker) [inline go.mod:32]: Not an expert on how go deps are resolved, but grafana-plugin-sdk-go v0.196.0 already points to kin-openapi 0.120, no? Is this line necessary?

**Review by sunker — APPROVED** [review]
Looks good to me. Just have a question.

**Pull request merged**